controller/controller.go

```diff
@@ -272,6 +349,14 @@ func deleteSchedulerHelper(logger logrus.FieldLogger, mr *models.MixedMetricsRep
 		}
 	}
 
+	err = mr.WithSegment(models.SegmentDelete, func() error {
```

Comment:
That makes sense.
I changed it because when I deleted a scheduler, the namespace was recreated without services and pods. I thought it was because the line on DB was not deleted, but actually just moving CreateNamespaceIfNecessary solved the problem. 
I will move it back.